src/sec.c

```diff
@@ -23,11 +23,11 @@ sec_bin_init(sec_bin_t *bin) {
 bool
 sec_init(tsdn_t *tsdn, sec_t *sec, base_t *base, pai_t *fallback,
     const sec_opts_t *opts) {
-	size_t max_alloc = opts->max_alloc & PAGE_MASK;
-	pszind_t npsizes = sz_psz2ind(max_alloc);
-	if (sz_pind2sz(npsizes) > opts->max_alloc) {
-		npsizes--;
-	}
+	assert(opts->max_alloc > 0);
+
+	size_t max_alloc = opts->max_alloc & ~PAGE_MASK;
```

Comment:
Maybe add a wrapper / macro for the `& ~` operation? around here: https://github.com/jemalloc/jemalloc/blob/dev/include/jemalloc/internal/pages.h#L14

Reply:
Good suggestion, was thinking about it myself. Applied.